### PR TITLE
feat:add-condition-to-pvc-creation

### PR DESCRIPTION
**Pull request by vicops**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -171,6 +171,7 @@ resource "kubernetes_deployment" "this" {
 #####
 
 resource "kubernetes_persistent_volume_claim" "this" {
+  count            = var.create_pvc ? 1 : 0
```
Comment (gladavius) [inline main.tf:174]: this condition should be based on the already existing variable enabled_localstorage

Basically if you want to use a local storage for grafana you will need 2 things : 
- 1 pvc
- 1 volume mount in the deployment

So these 2 resources are 2 faces of the same coin and that's why they to be based on the same condition.

Does it make sense?

File: variables.tf [hunk 2]
```diff
@@ -283,6 +283,13 @@ variable "pvc_labels" {
   type        = map(string)
 }
 
+variable "create_pvc" {
```
Comment (gladavius) [inline variables.tf:286]: see my previous comment, this variable can be removed

**Review by gladavius — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (vicops) [reply main.tf]: Yes David. I get your point. I will still try some few things

File: main.tf [hunk 3]
```diff
@@ -149,7 +149,7 @@ resource "kubernetes_deployment" "this" {
           content {
             name = "grafana-volume"
             persistent_volume_claim {
-              claim_name = kubernetes_persistent_volume_claim.this.metadata.0.name
+              claim_name = var.enabled_localstorage ? kubernetes_persistent_volume_claim.this[0].metadata.0.name : null
```
Comment (gladavius) [inline main.tf:152]: you don't need to add a new condition here, the dynamic block is already conditionned by this variable enabled_localstorage

**Review by gladavius — CHANGES_REQUESTED** [review]

**Review by gladavius — APPROVED** [review]

**Pull request merged**